---
title: "Visualizing LIME Results"
author: "Shay Iyer"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(igraph)
library(tidygraph)
library(ggraph)
set_graph_style()
```

## run viewingdata.Rmd to load all datasets

```{r}
allmaps <- rbind(baz2b.all, cdh5.all, celf1.all, foxp1.all, nrp1.all, nrp2.all, srsf1.all, yap1.all) 
rm(baz2b.all, cdh5.all, celf1.all, foxp1.all, nrp1.all, nrp2.all, srsf1.all, yap1.all) 
```

# what do i want to show: isoform transcripts with high splicing activity


# Making network graphs 

```{r making matrices and graphs out of data}
baz2b.edgelist <- baz2b.all %>% dplyr::select(eventID, transcript_id) %>% unique()
baz2b.edgelist <- as.matrix(baz2b.edgelist)
baz2b.g <- graph.edgelist(baz2b.edgelist, directed = T)

cdh5.edgelist <- cdh5.all %>% dplyr::select(eventID, transcript_id) %>% unique()
cdh5.edgelist <- as.matrix(cdh5.edgelist)
cdh5.g <- graph.edgelist(cdh5.edgelist, directed = T)

celf1.edgelist <- celf1.all %>% dplyr::select(eventID, transcript_id) %>% unique()
celf1.edgelist <- as.matrix(celf1.edgelist)
celf1.g <- graph.edgelist(celf1.edgelist, directed = T)

foxp1.edgelist <- foxp1.all %>% dplyr::select(eventID, transcript_id) %>% unique()
foxp1.edgelist <- as.matrix(foxp1.edgelist)
foxp1.g <- graph.edgelist(foxp1.edgelist, directed = T)

nrp1.edgelist <- nrp1.all %>% dplyr::select(eventID, transcript_id) %>% unique()
nrp1.edgelist <- as.matrix(nrp1.edgelist)
nrp1.g <- graph.edgelist(nrp1.edgelist, directed = T)

nrp2.edgelist <- nrp2.all %>% dplyr::select(eventID, transcript_id) %>% unique()
nrp2.edgelist <- as.matrix(nrp2.edgelist)
nrp2.g <- graph.edgelist(nrp2.edgelist, directed = T)

srsf1.edgelist <- srsf1.all %>% dplyr::select(eventID, transcript_id) %>% unique()
srsf1.edgelist <- as.matrix(srsf1.edgelist)
srsf1.g <- graph.edgelist(srsf1.edgelist, directed = T)

yap1.edgelist <- yap1.all %>% dplyr::select(eventID, transcript_id) %>% unique()
yap1.edgelist <- as.matrix(yap1.edgelist)
yap1.g <- graph.edgelist(yap1.edgelist, directed = T)

rm(baz2b.all, cdh5.all, celf1.all, foxp1.all, nrp1.all, nrp2.all, srsf1.all, yap1.all)
```

```{r baz2b}
plot(baz2b.g, layout = layout_in_circle(baz2b.g), edge.arrow.size = 0.25, label.cex = 0.75, vertex.size = 10) + title(main = "BAZ2B interaction network")

V(baz2b.g)[degree(baz2b.g, mode = 'in') >= 5]
# ENST00000392783.6 ENST00000437839.5
```

```{r celf1}
plot(celf1.g, layout = layout_in_circle(celf1.g), edge.arrow.size = 0.25, label.cex = 0.75, vertex.size = 10) + title(main = "CELF1 interaction network")

V(celf1.g)[degree(celf1.g, mode = 'in') >= 5]
```

```{r foxp1}

plot(foxp1.g, layout = layout_in_circle(foxp1.g), edge.arrow.size = 0.25, label.cex = 0.75, vertex.size = 10) + title(main = "foxp1 interaction network")

V(foxp1.g)[degree(foxp1.g, mode = 'in') >= 5]
```

```{r nrp1}

nrp1.g <- graph.edgelist(nrp1.edgelist, directed = T)
plot(nrp1.g, layout = layout_in_circle(nrp1.g), edge.arrow.size = 0.25, label.cex = 0.75, vertex.size = 10) + title(main = "nrp1 interaction network")

V(nrp1.g)[degree(nrp1.g, mode = 'in') >= 3]
```

```{r nrp2}

plot(nrp2.g, layout = layout_in_circle(nrp2.g), edge.arrow.size = 0.25, label.cex = 0.75, vertex.size = 10) + title(main = "nrp2 interaction network")

V(nrp2.g)[degree(nrp2.g, mode = 'in') >= 2]
```

```{r srsf1}

plot(srsf1.g, layout = layout_in_circle(srsf1.g), edge.arrow.size = 0.25, label.cex = 0.75, vertex.size = 10) + title(main = "srsf1 interaction network")

V(srsf1.g)[degree(srsf1.g, mode = 'in') >= 1]
```

```{r yap1}

plot(yap1.g, layout = layout_in_circle(yap1.g), edge.arrow.size = 0.25, label.cex = 0.75, vertex.size = 10) + title(main = "yap1 interaction network")

V(yap1.g)[degree(yap1.g, mode = 'in') >= 5]
```

```{r cdh5}

## modify below code
plot(cdh5.g, layout = layout_in_circle(cdh5.g), edge.arrow.size = 0.25, label.cex = 0.75, vertex.size = 10) + title(main = "cdh5 interaction network")


V(cdh5.g)[degree(cdh5.g, mode = 'in') >= 5]
# ENST00000392783.6 ENST00000437839.5 
```

```{r i went crazy}
allmaps2 <- allmaps %>% mutate(
  eventID_general = str_sub(eventID, end=-5)
)

allmaps.exc <- allmaps2 %>% filter(grepl("exc", eventID))

allmaps.both <- allmaps2 %>% filter(eventID_general %in% allmaps.exc$eventID_general)

events <- allmaps.both$eventID_general %>% unique()

events

allsplicing <- allsplicing %>% 
  mutate(
    eventID_general = str_sub(incID, end=-5)
  )
allsplicing.both <- allsplicing %>% filter(eventID_general %in% events)

ec.foxp1.expressed.known <- c("ENST00000649528.1", "ENST00000614176.5", "ENST00000649695.1", "ENST00000648748.1", "ENST00000650387.1", "ENST00000650231.1")

wtc11.foxp1.expressed.known <- c("ENST00000649528.1", "ENST00000650231.1")

ec.foxp1.q <- read_tsv("/Volumes/sheynkman/projects/shay_thesis/input-data/02_long-read-EC-WTC11-1_input/02_quantification/EC-FOXP1.tsv", col_names = FALSE) %>% as.data.frame()

allmaps.both.yap1 <- allmaps.both %>% filter(gene_name == "YAP1")

allmaps.both.yap1$transcript_id %>% unique()

```


To do

1. Find events where inclusion and exclusion forms mapped to different isoforms

2. For those genes, find isoforms that were actually expressed in the sample; calculate TPM

3. for both conditions, check if there is an isoform expressed in WTC 11 that has exon 18, isoform expressed in EC that has exon 18b...?

```{r}
allmaps <- allmaps %>% mutate(
  "eventID_general" = str_sub(eventID, end=-5)
)

allmaps.exc <- allmaps %>% filter(grepl("exc", eventID))
allmaps.exc$eventID %>% unique()
exclusions <- c("mxe.6043", "a5ss.3558", "a5ss.3557", "mxe.9065", "mxe.9066", "mxe.5476", "a5ss.7826", "a5ss.7827")

allmaps.both <- allmaps %>% filter(eventID_general %in% exclusions)

allmaps.both %>% group_by(eventID_general) %>% summarize(n = n()) %>% filter(n > 1) %>% select(eventID_general) %>% as.vector()

both <- c("a5ss.3558", "a5ss.7826", "a5ss.7827", "mxe.6043", "mxe.9065", "mxe.9066")

allmaps.both <- allmaps %>% filter(eventID_general %in% both)
```

```{r}
allmaps.both %>% select(gene_name) %>% unique()
```

```{r}
ec.quant<- read_tsv("/Volumes/sheynkman/projects/shay_thesis/input-data/02_long-read-EC-WTC11-1_input/02_quantification/EC.tsv") %>% as.data.frame()
wtc11.quant <- read_tsv("/Volumes/sheynkman/projects/shay_thesis/input-data/02_long-read-EC-WTC11-1_input/02_quantification/WTC11-1.tsv") %>% as.data.frame()
```

```{r}
ec.total = sum(ec.quant$rep1ENCSR148IIG)
wtc11.total = sum(wtc11.quant$rep1ENCSR507JOF)

ec.quant <- ec.quant %>% mutate(
  TPM = (rep1ENCSR148IIG/ec.total)*1000000
)

ec.quant <- ec.quant %>% rename(
  "transcript_id" = annot_transcript_id
)

wtc11.quant <- wtc11.quant %>% mutate(
  TPM = (rep1ENCSR507JOF/wtc11.total)*1000000
)

wtc11.quant <- wtc11.quant %>% rename(
  "transcript_id" = annot_transcript_id
)
```

```{r}

allmaps.both.foxp1 <- allmaps.both %>% filter(gene_name == "FOXP1")
allmaps.both.nrp2 <- allmaps.both %>% filter(gene_name == "NRP2")
allmaps.both.yap1 <- allmaps.both %>% filter(gene_name == "YAP1")
# allmaps.both <- allmaps %>% filter(eventID_general %in% both)
```

```{r}
allmaps.both.foxp1 <- left_join(allmaps.both.foxp1, ec.quant, by = "transcript_id")

allmaps.both.foxp1 <- left_join(allmaps.both.foxp1, wtc11.quant, by = "transcript_id")
```

```{r}
allmaps.both.foxp1 <- allmaps.both.foxp1 %>% select(c(gene_id, gene_name, strand, eventID, event_dpsi, transcript_id, type, eventID_general, rep1ENCSR148IIG, rep1ENCSR507JOF, TPM.x, TPM.y))
```

```{r}

ec.annot <- rtracklayer::import("/Volumes/sheynkman/projects/shay_thesis/input-data/02_long-read-EC-WTC11-1_input/01_annotation/EC-candidates.gtf") %>% as.data.frame() 
wtc11.annot <- rtracklayer::import("/Volumes/sheynkman/projects/shay_thesis/input-data/02_long-read-EC-WTC11-1_input/01_annotation/WTC11-1-candidates.gtf") %>% as.data.frame()
```

```{r}

ec.annot.exons <- ec.annot %>% filter(type == "exon")
wtc11.annot.exons <- wtc11.annot %>% filter(type == "exon")
concatenate_exon_values <- function(df) {
  # initialize an empty string to store the concatenated values
  exon_values <- ""
  print("initialized exon value")
  # loop through each row of the data frame
  for (i in 1:nrow(df)) {
    # extract the exon start and end values for the current row
    start <- df[i, "start"]
    end <- df[i, "end"]
    
    # concatenate the exon start and end values to the string
    exon_values <- paste0(exon_values, start, "-", end, ":")
  }
  print("loaded exon string")
  # return the concatenated string
  return(exon_values)
}

grouped <- ec.annot %>% group_by(transcript_id)
ec.annot.squished <- summarize(grouped, exon_values = concatenate_exon_values(grouped))

```

